src/common/logger.ts

```diff
@@ -0,0 +1,126 @@
+import envPaths from 'env-paths'
+import { join } from 'path'
+import * as winston from 'winston'
+
+/**
+ * Winston Base Log Levels for NPM
+ * {
+ *    error: 0,
+ *    warn: 1,
+ *    info: 2,
+ *    http: 3,
+ *    verbose: 4,
+ *    debug: 5,
+ *    silly: 6
+ * }
+ */
+
+export class EntropyLogger {
+  protected context: string
+  protected endpoint: string
+  private winstonLogger: winston.Logger
+
+  constructor (context: string, endpoint: string) {
+    this.context = context
+    this.endpoint = endpoint
+
+    let format = winston.format.combine(
+      // Add timestamp key: { timestamp: 'YYYY-MM-DD HH:mm:ss.SSS' }
+      winston.format.timestamp({
+        format: 'YYYY-MM-DD HH:mm:ss.SSS',
+      }),
+      // If message is instanceof Error, log Error's message property and stack
+      winston.format.errors({ stack: true }),
+      // Allows for string interpolation tokens '%s' in message with splat key values
+      // Ex. { message: 'my message %s', splat: ['test'] } -> { message: 'my message test' }
+      winston.format.splat(),
+      // Uses safe-stable-stringify to finalize full object message as string
+      // (prevents circular references from crashing)
+      winston.format.json(),
+    );
+
+    if (process.env.NODE_ENV === 'test') {
+      format = winston.format.combine(
+        format,
+        winston.format.colorize({ level: true }),
+        winston.format.printf(info => {
+          let message = typeof info.message === 'object' ? JSON.stringify(info.message) : info.message;
```

Comment:
Sometimes stringify output being all in one line is brutal, and spreading it over more lines, indented is nice.
Obvs takes up more space, not sure impact on logging style here tho

Code that would do that:
```suggestion
          let message = typeof info.message === 'object' ? JSON.stringify(info.message, null, 2) : info.message;
```